curriculum/challenges/english/25-front-end-development/lab-book-organizer/67172b43f84bcd2dec238a3d.md

```diff
@@ -0,0 +1,163 @@
+---
+id: 67172b43f84bcd2dec238a3d
+title: Build a Book Organizer
+challengeType: 14
+dashedName: build-a-book-organizer
+---
+
+# --description--
+
+**Objective:** Fulfill the user stories below and get all the tests to pass to complete the lab. 
+
+**User Stories:**
+
+1. You should have an array of objects named `books` where each object in the array should have a string `title`, another string `authorName`, and a number `releaseYear`. 
+
+2. Your `books` array should have a minimum of three objects.
+
+3. You should have a callback function named `sortByYear` that accepts two books as parameter for sorting the array.
+
+4. The `sortByYear` function should return `-1` if the `releaseYear` of the first book is smaller than that of the second book.
+
+5. The `sortByYear` function should return `1` if the `releaseYear` of the first book is bigger than that of the second book.
+
+6. The `sortByYear` function should return `0` for all other scenarios.
+
+7. You should filter out books written after a certain year such as 1950 from the `books` array and save the filtered array in a new array named `filteredBooks`.
+
+8. You should sort the books in the `books` array according to their `releaseYear` in ascending order and save the sorted array in a new array named `filteredBooksSorted`.
```

Comment:
```suggestion
8. You should sort the books in the `filteredBooks` array according to their `releaseYear` in ascending order and save the sorted array in a new array named `filteredBooksSorted`.
```

I think you mean this here? You could also maybe add to "use your sortByYear function".